developer/src/kmc-convert/src/keylayout-to-kmn/keylayout-to-kmn-converter.ts

```diff
@@ -353,49 +412,49 @@ export class KeylayoutToKmnConverter {
 
               // Data of Block Nr 1  .......................................................................................................................................................................
               // create array[Keycode,Keyname,action id,actionIndex,output] and array[Keyname,action id,behaviour,modifier,output] .........................................................................
-              /*  eg: ['49','K_SPACE','a0','0','Â'] */    const b1_keycode_arr: string[][] = this.get_KeyActionOutput_array__From__ActionStateOutput_array(jsonObj, b6_actionId_arr);
-              /*  eg: ['K_SPACE','a0','0','NCAPS','Â'] */ const b1_modifierKey_arr: string[][] = this.get_KeyMBehaviourModOutputArray__from__KeyActionBehaviourOutput_array(jsonObj, b1_keycode_arr, isCapsused);
-                // ...........................................................................................................................................................................................
-
-                for (let n1 = 0; n1 < b2_prev_deadkeyModifier_arr.length; n1++) {
-                  for (let n2 = 0; n2 < b2_prev_deadkeyModifier_arr[n1].length; n2++) {
-                    for (let n3 = 0; n3 < b2_prev_deadkey_arr.length; n3++) {
-                      for (let n4 = 0; n4 < b4_deadkeyModifier_arr.length; n4++) {
-                        for (let n5 = 0; n5 < b4_deadkeyModifier_arr[n4].length; n5++) {
-                          for (let n6 = 0; n6 < b4_deadkey_arr.length; n6++) {
-                            for (let n7 = 0; n7 < b1_modifierKey_arr.length; n7++) {
-
-                              rule_obj = new Rule(
+              /*  eg: ['49','K_SPACE','a0','0','Â'] */    const b1_keycode_obj: allArrayContents_object[] = this.get_KeyActionOutput_array__From__ActionStateOutput_array(jsonObj, b6_actionId_obj);
+              /*  eg: ['K_SPACE','a0','0','NCAPS','Â'] */ const b1_modifierKey_obj: allArrayContents_object[] = this.get_KeyMBehaviourModOutputArray__from__KeyActionBehaviourOutput_array(jsonObj, b1_keycode_obj, isCapsused);
+                  // ...........................................................................................................................................................................................
+
+                  for (let n1 = 0; n1 < b2_prev_deadkeyModifier_obj.length; n1++) {
+                    for (let n2 = 0; n2 < b2_prev_deadkeyModifier_obj[n1].length; n2++) {
+                      for (let n3 = 0; n3 < b2_prev_deadkey_obj.length; n3++) {
+                        for (let n4 = 0; n4 < b4_deadkeyModifier_obj.length; n4++) {
+                          for (let n5 = 0; n5 < b4_deadkeyModifier_obj[n4].length; n5++) {
+                            for (let n6 = 0; n6 < b4_deadkey_obj.length; n6++) {
+                              for (let n7 = 0; n7 < b1_modifierKey_obj.length; n7++) {
```

Comment:
Loops that nest this deeply are a sign that refactoring is needed. We should be breaking this down into separate functions.